src/randoop/operation/ArrayCreation.java

```diff
@@ -185,29 +211,48 @@ public boolean equals(Object o) {
     return true;
   }
 
+  /**
+   * {@inheritDoc}
+   * Creates string of the form
+   *   TYPE[NUMELEMS]
+   * where TYPE is the type of the array, and NUMELEMS is the number of elements.
+   * 
+   * Example:
+   *   int[3]
+   *   
+   * @return string descriptor for array creation.
+   */
+  @Override
   public String toParseableString() {
     return elementType.getName() + "[" + Integer.toString(length) + "]";
   }
 
   /**
-   * A string representing this array declaration. The string is of the form:
-   * 
-   * TYPE[NUMELEMS]
-   * 
-   * Where TYPE is the type of the array, and NUMELEMS is the number of elements.
-   * 
-   * Example:
-   * 
-   * int[3]
+   * Parses an array declaration in a string descriptor in the form generated
+   * by {@link ArrayCreation#toParseableString()}.
+   * @see OperationParser#parse(String)
    * 
+   * @param str  the string to be parsed for the {@code ArrayCreation}.
+   * @return the {@code ArrayCreation} object for the string.  
+   * @throws OperationParseException if string does not have expected form.
    */
-  public static Operation parse(String str) {
+  public static Operation parse(String str) throws OperationParseException {
     int openBr = str.indexOf('[');
     int closeBr = str.indexOf(']');
     String elementTypeStr = str.substring(0, openBr);
     String lengthStr = str.substring(openBr + 1, closeBr);
-    Class<?> elementType = Reflection.classForName(elementTypeStr);
+    try {
+    Class<?> elementType = TypeNames.getTypeForName(elementTypeStr);
```

Comment:
I would personally restrict the scope of the try...catch form to just this one statement rather than all three, but it's not a disaster to include the two extra statements.